core/commands/dag/dag.go

```diff
@@ -307,6 +307,24 @@ The output of blocks happens in strict DAG-traversal, first-seen, order.
 		// if err := car.Write(pipeW); err != nil {}
 
 		pipeR, pipeW := io.Pipe()
+		// this is so convoluted because we:
+		// - want to close the readpipe on error so that the goroutine dies
+		// - can not do so on success, because during a no-deamon process
+		//   scenario the PostRun will not be able to grab the EOF from the
+		//   now-closed pipe
```

Comment:
Interesting. That means that, when no daemon is run, we'll end up returning from `Run` before we're done writing the CAR.

This feels like a problem with the commands library.

Reply:
Note - I ran abut a hundred tests with sha256sum on the receiving end: I never got a "truncated run", which makes me think this is strictly an EOF detection thing

Reply:
Hm. That's odd. Actually, I wouldn't even expect this error. We shouldn't get past the `<-errCh` line unless we finish writing and close the pipe, or we hit an error.

Reply:
I just force-pushed removing this cap commit ( which puts everything the way it was )
I am fixing up import / merging / fixing sharness
Will dig into the error above a bit later